src/mapper/pkg/metricexporter/otel_edge_metric.go

```diff
@@ -0,0 +1,93 @@
+package metricexporter
+
+import (
+	"context"
+	"time"
+
+	"github.com/otterize/network-mapper/src/mapper/pkg/config"
+	sharedconfig "github.com/otterize/network-mapper/src/shared/config"
+	"github.com/spf13/viper"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc"
+	"go.opentelemetry.io/otel/exporters/stdout/stdoutmetric"
+	"go.opentelemetry.io/otel/metric"
+	sdk "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.21.0"
+)
+
+type OtelEdgeMetric struct {
+	meterProvider *sdk.MeterProvider
+	counter       metric.Int64Counter
+}
+
+func newResource() (*resource.Resource, error) {
+	return resource.Merge(resource.Default(),
+		resource.NewWithAttributes(semconv.SchemaURL,
+			semconv.OTelLibraryName("otterize/network-mapper"),
+		))
+}
+
+const ClientAttributeName = "client"
+const ServerAttributeName = "server"
+
+func newMeterProvider(ctx context.Context, res *resource.Resource) (*sdk.MeterProvider, error) {
```

Comment:
This is ok for a prototype, but I would suggest that creating the MeterProvider should be a concern of main, so other components can make metrics if needed, and this uses the `otel.GetMeterProvider`.

Main (or a telemetry pkg) would hold this logic, and when a command starts it would `otel.SetMeterProvider`, and that would enable adding telemetry throughout the tools.

Reply:
Makes sense, for now it's separate because (somewhat uniquely) OTel isn't instrumenting the code itself, but generating a metric from a signal received by the code.